src/Swarm/Language/Text/Markdown.hs

```diff
@@ -230,13 +269,16 @@ chunksOf n = chop (splitter True n)
   cut :: Bool -> Int -> StreamNode -> (StreamNode, StreamNode)
   cut start i tn =
     let (con, t) = unStream tn
-     in case splitWordsAt i (T.words t) of
+        endSpace = T.takeWhileEnd isSpace t
+        startSpace = T.takeWhile isSpace t
+        twords = T.words t & _head %~ (startSpace <>) & _last %~ (<> endSpace)
+     in case splitWordsAt i twords of
           ([], []) -> (con "", con "")
           ([], ws@(ww : wws)) ->
             both (con . T.unwords) $
               -- In case single word (e.g. web link) does not fit on line we must put
               -- it there and guarantee progress (otherwise chop will cycle)
-              if start then ([ww], wws) else ([], ws)
+              if start then ([T.take i ww], T.drop i ww : wws) else ([], ws)
```

Comment:
Can we utilize `T.splitAt i ww` here?

Reply:
Maybe, but I doubt it would make the code simpler or more performant.

Reply:
Sure, it's up to your preference.  I suppose it could look like this.
```suggestion
              if start then bimap pure (: wws) $ T.splitAt i ww else ([], ws)
```

Reply:
Thanks, I was about to try write it myself, but this is definitely harder to read for me. 😆